spec/requests/maintenance_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require "rails_helper"
+
+RSpec.describe MaintenanceController, type: :request do
+  let(:page) { Capybara::Node::Simple.new(response.body) }
+  let(:user) { FactoryBot.create(:user) }
+
+  before do
+    sign_in user
+  end
+
+  describe "when maintenance mode is enabled" do
+    before do
+      allow(FeatureToggle).to receive(:maintenance_mode_enabled?).and_return(true)
+    end
+
+    context "when a user visits a page other than the maintenance page" do
+      before do
+        get "/lettings-logs"
+      end
+
+      it "redirects the user to the maintenance page" do
+        expect(response).to redirect_to(service_unavailable_path)
+        follow_redirect!
+        expect(page).to have_content("Sorry, the service is unavailable")
+      end
+
+      it "the cookie banner is visible" do
```

Comment:
I could imagine it being sensible to move these cookie banner tests into another file (e.g. `spec/views/layouts/application_layout_spec.rb`, although I'm not convinced that is the right place) but I thought they were probably fine here. Happy to be persuaded otherwise though.